filecoin/functions/handle-cron-tick.js

```diff
@@ -18,28 +18,48 @@ Sentry.AWSLambda.init({
 const AWS_REGION = process.env.AWS_REGION || 'us-west-2'
 
 export async function handleCronTick () {
-  const { did, pieceTableName, agentMessageBucketName, agentIndexBucketName, aggregatorDid, storefrontProof } = getEnv()
+  const { did, pieceTableName, agentMessageBucketName, agentIndexBucketName, aggregatorDid } = getEnv()
   const { PRIVATE_KEY: privateKey } = Config
 
+  let aggregatorProof
+  try {
+    aggregatorProof = Config.AGGREGATOR_SERVICE_PROOF
+  } catch (error) {
+    // AGGREGATOR_SERVICE_PROOF is not set for this environment
+    aggregatorProof = undefined
+  }
+
   // create context
-  let id = getServiceSigner({
-    privateKey
+  const storefrontSigner = getServiceSigner({
+    did,
+    privateKey,
   })
-  const storefrontServiceProofs = []
-  if (storefrontProof) {
-    const proof = await Proof.parse(storefrontProof)
-    storefrontServiceProofs.push(proof)
-  } else {
-    // if no proofs, we must be using the service private key to sign
-    id = id.withDID(DID.parse(did).did())
+
+  const aggregatorServiceProofs = []
+  if (aggregatorProof) {
+    const proof = await Proof.parse(aggregatorProof)
+    aggregatorServiceProofs.push(proof)
   }
 
+  const aggregatorServiceSigner = getServiceSigner({
+    did: aggregatorDid,
+    privateKey: privateKey,
+  })
```

Comment:
Hmm, ok this reads weirdly. I think what we want to do is require a proof if the service DID is different to the aggregator DID:

```js
if (did !== aggregatorDid && !aggregatorProof) {
  throw new Error(`proof(s) required for ${did} to use aggregator ${aggregatorDid}`)
}
```

Then you remove this and always use the `storefrontSigner` as the issuer.

Reply:
You are right! We can't use the service `did:web` to sign with, it MUST be the aggregator `did:web`. I was thinking if we're using the same private key that the `did:web` would also match, however that is not always the case (it might be `did:web:up.storacha.network`).